colossalai/utils/profiler/common.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+from .comm_profiler import CommProfiler
+from typing import List
+from torch.utils.tensorboard import SummaryWriter
+
+
+class ProfilerEvent(Enum):
+    '''
+    Event types to profile
+    '''
+    COMM = CommProfiler
+
+
+class ProfilerContext(object):
+
+    def __init__(self, events: List[ProfilerEvent]):
+        self.events = list()
+        for event_cls in events:
+            self.events.append(event_cls())
+
+    def __enter__(self):
+        for e in self.events:
+            e.enable()
+
+    def __exit__(self):
+        for e in self.events:
+            e.disable()
+
+    def to_tensorboard(self, writer: SummaryWriter):
+        for e in self.events:
+            e.to_tensorboard(writer)
+
+    def to_file(self, filename: str):
+        for e in self.events:
+            e.to_file(filename)

```

Comment:
multiple profilers write into the same file?

Reply:
How about provide a path and add 'profilerevent_name.log' as a suffix

Reply:
One way to handle this is to set the format as {output_path}/{prefix}_{profiler.__class__.__name__}_{timestamp}.log. This method requires changing the parameter filename to otuput_path